var/spack/repos/builtin/packages/pika/package.py

```diff
@@ -91,6 +90,11 @@ class Pika(CMakePackage, CudaPackage, ROCmPackage):
     conflicts("%clang@:8", when="@0.2:")
     conflicts("+stdexec", when="cxxstd=17")
     conflicts("cxxstd=23", when="^cmake@:3.20.2")
+    # Adds conflict with C++ standard > 17 and cuda < 12
+    non_17_cxxstd = list(cxxstds)
+    non_17_cxxstd.remove("17")
+    for standard in non_17_cxxstd:
+        conflicts("cxxstd="+standard, when="^cuda@:11")
```

Comment:
Just to stay consistent with the existing naming conventions in the package:
```suggestion
    # Adds conflict with C++ standard > 17 and cuda < 12
    non_17_cxxstds = list(cxxstds)
    non_17_cxxstds.remove("17")
    for cxxstd in non_17_cxxstds:
        conflicts("cxxstd={0}".format(cxxstd), when="^cuda@:11")
```

Not sure what the most pythonic way of doing is, but maybe also:
```suggestion
    # CUDA version <= 11 does not support C++20 and newer
    for cxxstd in filter(lambda x: x != "17", cxxstds):
        conflicts("cxxstd={0}".format(cxxstd), when="^cuda@:11")
```

Reply:
you are also encouraged to use `f"cxxstd={cxxstd}"` f-strings